src/Voron/Impl/Paging/CryptoPager.cs

```diff
@@ -221,6 +221,15 @@ internal override void UnprotectPageRange(byte* start, ulong size, bool force =
             return buffer.Pointer;
         }
 
+        public override byte* AcquirePagePointerHeaderForDebug(IPagerLevelTransactionState tx, long pageNumber, PagerState pagerState = null)
```

Comment:
I would rather return a `Span` here, clearly limiting the scope of what you can read.
Or just the `PageHeader*`

Reply:
Better yet, maybe have the _caller_ provide a span and we'll copy to there?

Reply:
Done